src/status.rs

```diff
@@ -111,31 +114,40 @@ pub async fn is_finalized(sentry: &SentryApi, campaign: &Campaign) -> Result<IsF
 
 pub async fn get_status(sentry: &SentryApi, campaign: &Campaign) -> Result<Status, Error> {
     // continue only if Campaign is not Finalized
+    let leader = campaign.channel.spec.validators.leader();
     let leader_la = match is_finalized(sentry, campaign).await? {
         IsFinalized::Yes { reason, balances } => return Ok(Status::Finalized(reason, balances)),
         IsFinalized::No { leader } => leader,
     };
+    let leader_la_clone = match is_finalized(sentry, campaign).await? {
+        IsFinalized::Yes { reason, balances } => return Ok(Status::Finalized(reason, balances)),
+        IsFinalized::No { leader } => leader,
+    };
 
     let follower = campaign.channel.spec.validators.follower();
     let follower_la = sentry.get_last_approved(&follower).await?;
+    let follower_la_clone = sentry.get_last_approved(&follower).await?;
 
     // setup the messages for the checks
     let messages = Messages {
         last_approved: leader_la.last_approved,
         leader_heartbeats: get_heartbeats(leader_la.heartbeats),
         follower_heartbeats: get_heartbeats(follower_la.heartbeats),
+        leader_new_state: get_new_state(leader_la_clone.last_approved),
```

Comment:
all these can be removed. All this data can be extracted from the `leader_la` and `follower_la` so these are the things we need and we can perform all the checks with them using iterators.